src/data/decimal/decimalFn.ts

```diff
@@ -0,0 +1,72 @@
+import { assert } from '../../utils/assert'
+import { PLDecimal } from './PLDecimal'
+
+export function plDecimal(strValue: string): PLDecimal {
+  const decimalObj = parseNumString(strValue)
+  return new PLDecimal(decimalObj.intValue, decimalObj.decimals)
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function assertNumeric(strValue: any): any {
+  return assert(strValue === '' || isNaN(strValue), `Invalid decimal number: "${strValue}"`)
+}
+
+export function isScientific(strValue: string): boolean {
+  const parts = strValue.split(/[eE]/)
+  if (parts.length === 2 && assertNumeric(parts[0]) && assertNumeric(parts[1])) {
+    const intValueRounded = Math.abs(parseInt(parts[0]))
```

Comment:
- It's ok to lose the fraction part. i only need to know if the number is between 1 and 9 (otherwise the number is not in a  scientific form)
- I would prefer not to use `parseFloat` because it is too smart: it can parse scientific numbers e.g. `1e3` even if that should already be excluded at `parts[0]`.

Reply:
I've rewritten the function using regexp.